src/pages/index.js

```diff
@@ -170,16 +242,187 @@ profileCard.setEventListeners();
+
+api
+  .getInitialCards()
+  .then((initialCards) => {
+    sectionRenderer = new Section(
+      {
+        items: initialCards,
+        renderer: (cardData) => {
+          const card = renderCard(cardData);
+          sectionRenderer.addItem(card);
+        },
+      },
+      cardListSelector
+    );
+    sectionRenderer.renderItems();
+    const cards = initialCards;
+  })
+  //need to figure out a way to take the ids and plug them into the url. Do i use an event listener of every card and everytime i click a specific one i get the id for it i do i just listen for every time?
+  .catch((err) => {
+    console.error("An error was found", err);
+  });
+
+/* -------------------------------------------------------------------------- */
+/*                                fetch request                               */
+/* -------------------------------------------------------------------------- */
+api
+  .userProfileInfo()
+  .then((res) => res.json())
+  .then((info) => {
+    console.log(info);
+    userInfo.setUserInfo({
+      name: info.name,
+      job: info.about,
+    });
+    console.log(userInfo);
+  })
+  .catch((err) => {
+    console.error("an error has occurred", err);
+  })
+  .finally(() => {
+    console.log("done");
+  });
+
+/* -------------------------------------------------------------------------- */
+/*                                card requests                               */
+/* -------------------------------------------------------------------------- */
+
+function handleLikeButton(card) {
+  if (card._isLiked) {
+    api
+      .removeLikeButtonRequest(card._id)
+      .then((res) => {
+        card.updateLikes(res.isLiked);
+      })
+      .catch((err) => {
+        console.error("an error has occurred", err);
+      })
+      .finally(() => {
+        console.log("done");
+      });
+  } else {
+    api
+      .likeButtonRequest(card._id)
+      .then((res) => {
+        card.updateLikes(res.isLiked);
+      })
+      .catch((err) => {
+        console.error("an error has occurred", err);
+      })
+      .finally(() => {
+        console.log("done");
+      });
+  }
+}
+
+/* -------------------------------------------------------------------------- */
+/*                            profile image request                            */
+/* -------------------------------------------------------------------------- */
+
+function setProfilePicture(linkInput) {
+  profileImage.src = linkInput;
+}
+
+const updateProfilePicture = new PopupWithForm({
+  popupSelector: "#change-avatar-modal",
+  handleFormSubmit: (inputValues) => {
+    const linkInput = inputValues.link;
+    console.log(linkInput);
+    api
+      .updateProfilePictureRequest({ linkInput: linkInput })
+      .then((res) => {
+        console.log(res);
+        setProfilePicture(linkInput);
+        updateProfilePicture.close();
+        // this is where the pushing of the link goes through to the src on the img
+      })
+      .catch((err) => {
```

Comment:
Good job that you `catch` possible errors at the end of server requests. You can simplify it with `.catch(console.error);` because `console.error` is meant to log errors